### PR TITLE
changes UnionRevisionLabelPodDeployer to deal with `nil` secrets

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

File: pkg/operator/encryption/deployer/unionrevisionedpod.go [hunk 1]
```diff
@@ -36,6 +36,9 @@ func (d *UnionRevisionLabelPodDeployer) DeployedEncryptionConfigSecret() (secret
 		if !converged || err != nil {
 			return nil, converged, err
 		}
+		if secret == nil {
```
Comment (p0lyn0mial) [inline pkg/operator/encryption/deployer/unionrevisionedpod.go]: I will use it in `OAS-O` - the custom `deployer` will not return any resources when the annotation was `set`.

Comment (sttts) [reply pkg/operator/encryption/deployer/unionrevisionedpod.go]: What do we see if the oauth-apiserver secret does not exist?

Comment (p0lyn0mial) [reply pkg/operator/encryption/deployer/unionrevisionedpod.go]: the same - https://github.com/openshift/library-go/blob/master/pkg/operator/encryption/deployer/revisionedpod.go#L116

Comment (p0lyn0mial) [reply pkg/operator/encryption/deployer/unionrevisionedpod.go]:
```
// if encryption is not enabled at this revision or the secret was deleted, we should not error
if errors.IsNotFound(err) {
	return nil, true, nil
}
```

Comment (sttts) [reply pkg/operator/encryption/deployer/unionrevisionedpod.go]: My thought was that we might want to block COAO from progressing if its secret exists, but CAO's secret does not.

Comment (p0lyn0mial) [reply pkg/operator/encryption/deployer/unionrevisionedpod.go:39]: if https://github.com/openshift/cluster-openshift-apiserver-operator/pull/348 makes sense then we need to merge this pull.

**Pull request closed without merging**